vendor/tests.py

```diff
@@ -0,0 +1,209 @@
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertFalse(Vendor.objects.filter(id=self.vendor.id).exists())
+
+    def test_get_vendor_list_no_vendors(self):
+        Vendor.objects.all().delete()
+        url = reverse('vendor-list-create')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, [])
+
+    def test_create_vendor_missing_data(self):
+        url = reverse('vendor-list-create')
+        incomplete_data = {"name": "Incomplete Vendor"}
+        response = self.client.post(url, incomplete_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_get_vendor_detail_does_not_exist(self):
+        url = reverse('vendor-retrieve-update-destroy', args=[999])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_vendor_does_not_exist(self):
+        url = reverse('vendor-retrieve-update-destroy', args=[999])
+        updated_data = {"name": "Updated Vendor"}
+        response = self.client.put(url, updated_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_vendor_does_not_exist(self):
+        url = reverse('vendor-retrieve-update-destroy', args=[999])
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class VendorEventListViewTest(BaseTestCase):
+    def test_get_vendors_for_event(self):
+        url = reverse('vendor-event-list', kwargs={'event_id': self.event_id})
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        
+        expected_data = VendorSerializer([self.vendor, self.vendor1], many=True).data
+        self.assertEqual(response.data, expected_data)
+    
+    def test_get_vendors_for_event_no_vendors(self):
+        Vendor.objects.all().delete()
+        url = reverse('vendor-event-list', kwargs={'event_id': self.event_id})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, [])
+
+class PhotoModelTestCase(BaseTestCase):
+    def test_photo_model(self):
+        self.assertEqual(self.photo.vendor, self.vendor)
+        self.assertEqual(str(self.photo), "https://example.com/path/to/your/image.jpg")
+
+    def test_photo_model_no_vendor(self):
+        with self.assertRaises(IntegrityError):
+            PhotoVendor.objects.create(
+                vendor=None,
+                image="https://example.com/path/to/your/image.jpg"
+            )
+
+    def test_photo_model_no_image(self):
+        with self.assertRaises(IntegrityError):
+            PhotoVendor.objects.create(
+                vendor=self.vendor,
+                image=None
```

Comment:
Does creating vendor means photo is also a requirement? I think we can leave it as optional?

Reply:
The `PhotoVendor` model represents the photos associated with a vendor. Therefore, each `Vendor` object will have an associated list of `PhotoVendor` objects. The reason for this specific naming convention is to distinguish between photos related to the `Vendor` and `Venue` models, as both models will have their own separate gallery features. Therefore, it can't be None.

Reply:
Noted, I think this is great way to keep things decoupled. Nice job!